### PR TITLE
docs: add tool to generate instance type docs

### PR DESCRIPTION
**Pull request by tzneal**
Fixes # 1329

**Description**

Displays both labels and the resources that we use for bin-packing after overhead calculations.
**How was this change tested?**

* Preview with netlify.

**Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [ ] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: hack/docs/instancetypes_gen_docs.go [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+weight: 100
+
+description: >
+  Evaluate Instance Type Resources
+---
+`)
+	fmt.Fprintln(f, "<!-- this document is generated from hack/docs/instancetypes_gen_docs.go -->")
+	fmt.Fprintln(f, `AWS instance types offer varying resources and can be selected by labels. In particular the
+ephemeral-storage is the default value for an AL2 family assuming no specific block device mappings have been configured.`)
+
```
Comment (bwagner5) [inline hack/docs/instancetypes_gen_docs.go:72]: Might be worth calling out the the resources have overhead taken out.  Although it looks like ephemeral storage does not and maybe not CPU either?  https://github.com/aws/karpenter/blob/205c4db5743f1f6824253e92304a5df2f875c90c/pkg/cloudprovider/aws/amifamily/al2.go#L94

Might also be worth calling out that pods are assuming the `aws-eni-limited-pod-density = true`

**Review by bwagner5 — COMMENTED** [review]
NICE! just one comment

File: hack/docs/instancetypes_gen_docs.go (hunk 1 above)
Comment (tzneal) [reply hack/docs/instancetypes_gen_docs.go:72]: SGTM

**Review by bwagner5 — APPROVED** [review]
lgtm! Big improvement! to docs!

**Pull request merged**